File: TaliExpress.Server/Models/Product.cs
using MongoDBService.Classes;

namespace TaliExpress.Server.Models
{
    public class Product : AMongoDBItem
    {
        public double Price { get; set; } = 0;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public int Stock { get; set; } = 0;
        public string SellerId { get; set; } = string.Empty;
        public bool IsAvailable { get; set; } = false;
        public double DiscountId { get; set; } = 0;
    }
}

File: Taliexpress.client/Classes/Common/Product.ts

export class Product {
  id: string = '';
  creationDate!: Date;
  name: string = '';
  description: string = '';
  amountInInvenotry: number = 0;
  userId: string = '';
  price: number = 0;
  //images: Image = new Image();
  categories: string[] = [];
  isAvailable: boolean = false;
  discountId: number = 0;
}
